src/pages/ReportsUpload/index.tsx

```diff
@@ -0,0 +1,341 @@
+import {
+    Button,
+    Modal,
+    ButtonVariant,
+    Bullseye,
+    EmptyState,
+    EmptyStateVariant,
+    EmptyStateIcon,
+    Title,
+    EmptyStateBody,
+    ProgressMeasureLocation,
+    Progress,
+    TitleLevel,
+    ProgressVariant,
+    EmptyStateSecondaryActions
+} from '@patternfly/react-core';
+import { VolumeIcon } from '@patternfly/react-icons';
+import { Formik } from 'formik';
+import * as Yup from 'yup';
+import * as uploadActions from '../../actions/UploadActions';
+import { GlobalState } from '../../models/state';
+import { RouterGlobalProps } from '../../models/router';
+import {
+    Main,
+    PageHeader,
+    PageHeaderTitle
+} from '@redhat-cloud-services/frontend-components';
+import { Upload } from '../../models';
+import './ReportsUpload.scss';
+import { UploadForm } from './UploadForm';
+
+interface FormValues {
+    file: string;
+    reportName: string;
+    reportDescription: string;
+    yearOverYearGrowthRatePercentage: number;
+    percentageOfHypervisorsMigratedOnYear1: number;
+    percentageOfHypervisorsMigratedOnYear2: number;
+    percentageOfHypervisorsMigratedOnYear3: number;
+    percentageOfHypervisorsMigratedSum: number // Sum of year1, year2, and year3
+}
+
+interface StateToProps {
+    file: File;
+    success: boolean | null;
+    error: string | null;
+    progress: number;
+    uploading: boolean
+}
+
+interface DispatchToProps {
+    uploadProgress: (progress: number) => void;
+    uploadRequest: (upload: Upload, config: {}) => void;
+    selectUploadFile: (file: File) => void;
+}
+
+interface Props extends StateToProps, DispatchToProps, RouterGlobalProps {
+}
+
+interface State {
+    showForm: boolean;
+    timeoutToRedirect: number;
+}
+
+export class ReportsUpload extends React.Component<Props, State> {
+
+    redirectTimer: any;
+    beforeUnloadHandler: any;
+
+    initialFormValue: FormValues;
+
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            showForm: true,
+            timeoutToRedirect: 3
+        };
+
+        this.initialFormValue = {
+            file: '',
+            reportName: '',
+            reportDescription: '',
+            yearOverYearGrowthRatePercentage: 5,
+            percentageOfHypervisorsMigratedOnYear1: 50,
+            percentageOfHypervisorsMigratedOnYear2: 30,
+            percentageOfHypervisorsMigratedOnYear3: 10,
+            percentageOfHypervisorsMigratedSum: 90
+        };
+
+        this.redirectTimer = null;
+    }
+
+    componentDidMount() {
+        this.beforeUnloadHandler = window.addEventListener('beforeunload', (event) => {
+            if (this.props.uploading) {
+                event.preventDefault();
+                return event.returnValue = 'Are you sure you want to close? The upload has not finished yet';
+            }
+
+            return event;
+        });
+    }
+
+    componentWillUnmount() {
+        removeEventListener('beforeunload', this.beforeUnloadHandler);
```

Comment:
I'd either put `window.` here or remove it from `addEventListener` above for consistency.

Reply:
I've put `window`. Thanks for the suggestion!